spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/AbstractStreamService.java

```diff
@@ -118,14 +122,15 @@ public StreamDefinition createStream(String streamName, String dsl, boolean depl
 					StringUtils.collectionToDelimitedString(errorMessages, "\n"));
 		}
 
-		this.streamDefinitionRepository.save(streamDefinition);
+		final StreamDefinition savedStreamDefintion = this.streamDefinitionRepository.save(streamDefinition);
+
 		if (deploy) {
 			this.deployStream(streamName, new HashMap<>());
 		}
 
 		auditRecordService.populateAndSaveAuditRecord(
 				AuditOperationType.STREAM, AuditActionType.CREATE,
-				streamDefinition.getName(), streamDefinition.getDslText());
+				streamDefinition.getName(), this.argumentSanitizer.sanitizeStream(savedStreamDefintion));
```

Comment:
I think that the `ArgumentSanitizer` should be pushed down into the `AuditRecordService` to encapsulate it better.  It would also remove duplication between the Skipper and non-Skipper StreamServices.

Reply:
Will address in #2477.  Merging this for now to avoid extra conflicts with adding argument sanitization for Tasks.

Reply:
merged as 0d207b945190fe8913078c5b06f66b343b9d6514